### PR TITLE
First working netlink event handling

### PR DESCRIPTION
**Pull request by rgwohlbold**
This PR changes our codebase to only use netlink for event handling.
We listen on `CMD_NEW_STATION` events from which we get two: One with the physical interface, e.g. wlp0s20f3. The other one contains the VLAN interface. Both events contain the MAC address of the station that connected.

Therefore, we can set up the station when we receive `CMD_NEW_INTERFACE` with the VLAN interface.

This PR should be functional, but needs some future fixups:

- [ ] Use netlink in cleanup
- [ ] Improve VLAN parsing
- [ ] RAII for netlink resources in Socket80211
- [ ] Improve naming of Socket and Socket80211
- [ ] Adapt Grafana

### CODE REVIEW

File: src/nl/Socket80211.cpp [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+#include "../metrics/MetricsManager.h"
+
+static std::string format_mac(uint8_t mac[6]) {
+    char formatted_mac[2 * 6 + 5 + 1];
+    snprintf(formatted_mac, sizeof(formatted_mac), "%02x:%02x:%02x:%02x:%02x:%02x", mac[0], mac[1], mac[2], mac[3], mac[4], mac[5]);
+    return formatted_mac;
+}
+
+static int interface_event_handler(struct nl_msg *msg, void *arg) {
+    auto *hdr = static_cast<genlmsghdr *>(nlmsg_data(nlmsg_hdr(msg)));
```
Comment (dasGoogle) [inline src/nl/Socket80211.cpp:24]: header?

File: src/nl/Socket80211.cpp [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+            ifindex = nla_get_u32(attrs[NL80211_ATTR_IFINDEX]);
+            char name[IFNAMSIZ];
+            if (if_indextoname(ifindex, name) == nullptr) {
+                GAFFALOG(DEBUG) << "NL80211_CMD_NEW_STATION mac=" << mac << " ifindex=" << ifindex << " with no associated name";
+            } else {
+                GAFFALOG(DEBUG) << "NL80211_CMD_NEW_STATION mac=" << mac << " ifindex=" << ifindex << " name=" << name;
+                std::string ifacename(name);
+                if (ifacename.rfind(VLAN_INTERFACE_PREFIX, 0) == 0) {
+                    socket->station_counter_received.Increment();
+                    // TODO: improve
```
Comment (dasGoogle) [inline src/nl/Socket80211.cpp:59]: Could we instead get the VLAN ID here?

File: src/nl/Socket80211.cpp [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+}
+
+
+nl::Socket80211::Socket80211(const std::chrono::duration<int>& timeout)
+    : station_counter_received(MetricsManager::get_instance().get_station_counter_received()) {
+    socket = nl_socket_alloc();
+    if (socket == nullptr) {
+        throw std::runtime_error(std::string("could not allocate netlink socket: ") + std::strerror(errno));
+    }
+    nl_socket_disable_seq_check(socket);
```
Comment (dasGoogle) [inline src/nl/Socket80211.cpp:75]: Should we do that here? 
We might want to check whether this works without this line or if this line has any negative effects

File: src/nl/Socket80211.cpp [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    int ret = genl_connect(socket);
+    if (ret < 0) {
+        throw std::runtime_error(std::string("could not connect to generic netlink family: ") + std::strerror(errno));
```
Comment (dasGoogle) [inline src/nl/Socket80211.cpp:78]: What exactly does this error mean? What is a family?

**Review by dasGoogle — CHANGES_REQUESTED** [review]
Just some minor nitpicks we talked about

File: src/nl/Socket80211.cpp (hunk 1 above)
Comment (dasGoogle) [reply src/nl/Socket80211.cpp:24]: Although, if it is NL-Lingo, we can keep it

Comment (rgwohlbold) [reply src/nl/Socket80211.cpp:24]: will fix :+1:

File: src/nl/Socket80211.cpp (hunk 3 above)
Comment (rgwohlbold) [reply src/nl/Socket80211.cpp:72]: We get an error: "Message sequence number mismatch", so it does not seem possible

Comment (rgwohlbold) [reply src/nl/Socket80211.cpp:72]: https://stackoverflow.com/questions/64669383/cannot-understand-this-message-sequence-mismatch-error

**Review by dasGoogle — APPROVED** [review]

**Pull request merged**